simpleratingbar/src/main/java/com/iarcuschin/simpleratingbar/SimpleRatingBar.java

```diff
@@ -673,6 +677,56 @@ private void setNewRatingFromTouch(float x, float y) {
       }
     }
   }
+  
+  @Override
+  protected Parcelable onSaveInstanceState() {
+      Parcelable superState = super.onSaveInstanceState();
+      SavedState savedState = new SavedState(superState);
+      savedState.rating = getRating();
+      return savedState;
+    }
+
+  @Override
+  protected void onRestoreInstanceState(Parcelable state) {
+      SavedState savedState = (SavedState) state;
+      super.onRestoreInstanceState(savedState.getSuperState());
+      setRating(savedState.rating);
```

Comment:
You are right. Albeit, I notice android's TextView class restores state the same way, so I wonder if that is the default expectation of onRestoreInstanceState

ref: http://grepcode.com/file/repository.grepcode.com/java/ext/com.google.android/android/5.1.1_r1/android/widget/TextView.java#3778